src/org/jitsi/impl/neomedia/rtp/sendsidebandwidthestimation/SendSideBandwidthEstimation.java

```diff
@@ -488,4 +566,144 @@ private synchronized void fireBandwidthEstimationChanged(
             second = b;
         }
     }
+
+    public class StatisticsImpl implements Statistics
+    {
+        private LossRegion currentState = null;
+
+        // looping over the same state is still a transition.
+        private long lastTransitionTimestampMs = -1;
+
+        // Current state statistics.
+        private long currentStateCumulativeDurationMs;
+        private int currentStateConsecutiveVisits;
+        private long currentStateStartBitrateBps;
+        private LongSummaryStatistics currentStateBitrateStatistics
+            = new LongSummaryStatistics();
+        private IntSummaryStatistics currentStateLossStatistics
+            = new IntSummaryStatistics();
+
+        // State duration statistics.
+        private final LongSummaryStatistics
+            lossFreeMsStats = new LongSummaryStatistics(),
+            lossDegradedMsStats = new LongSummaryStatistics(),
+            lossLimitedMsStats = new LongSummaryStatistics();
+
+        @Override
+        public void update(long nowMs)
+        {
+            synchronized (SendSideBandwidthEstimation.this)
+            {
+                update(nowMs, null);
+            }
+        }
+
+        // This method is synchronized on SendSideBandwidthEstimation.this.
```

Comment:
There's no synchronized block. Do you mean that it should always be called while holding the lock? If so, I would make the method synchronized. Or at least make it private.

Reply:
> Do you mean that it should always be called while holding the lock?

Yes.

> If so, I would make the method synchronized.

👍